requirements.in

```diff
@@ -4,11 +4,14 @@ docutils==0.20.1
 event-bus==1.0.2
 idna==3.4
 kaki==0.1.9
-Kivy @ https://github.com/kivy/kivy/archive/master.zip
+Kivy==2.2.1
```

Comment:
I think it's OK to build for `iOS`, but there is an issue when building for Android with the current 2.2.1 version thought 🤔 

- if `Kivy @ https://github.com/kivy/kivy/archive/master.zip` is working for both `iOS` and `Android` I would keep it as is, otherwise, I believe we need to give two different instructions or have commented lines here 🤔 what do you think?

Reply:
Looking into this... I'm not sure what's going on. I've tried to recreate the environment from scratch to make sure I get all the right/necessary settings, but now I'm starting to get a new set of issues. I will keep you posted.

Reply:
Ok, something strange is going on because when I use the `Kivy @ [https://github.com/kivy/kivy/archive/master.zip`](https://github.com/kivy/kivy/archive/master.zip%60) package I get this error when I just try to run it from the terminal with python: ` ModuleNotFoundError: No module named 'kivy.core.window._window_sdl2'`. Have you dealt with this?

I am assuming you can build it. Therefore, would you do `pip freeze > requirments.txt` with your lates venv that runs the app from the python terminal. Then send me that requirements.txt. Unfortunately, the current master still requires me to add pacakges and every combination of packages with  `Kivy @[https://github.com/kivy/kivy/archive/master.zip`](https://github.com/kivy/kivy/archive/master.zip%60) package will not even run on python for me.

Reply:
On my side I think I already got this `_window_sdl2` issue, solved by `brew upgrade`...

Below is what's generated by `pip freeze > requirements.txt`:
```
build==1.0.3
certifi==2023.7.22
charset-normalizer==3.2.0
click==8.1.7
docutils==0.20.1
event-bus==1.0.2
idna==3.4
kaki==0.1.9
Kivy @ https://github.com/kivy/kivy/archive/master.zip#sha256=8d7fea6be95e6b77462366f75f0f346571a94c953d5f1c13bbcfb50c7e3e78e3
Kivy-Garden==0.1.5
kivymd @ https://github.com/kivymd/KivyMD/archive/master.zip#sha256=e55b90474afa09a6c971f4ef1cdf07a4e028d7e565d8165179b7fc9b8f6b4d1b
materialyoucolor==1.2.0
packaging==23.2
Pillow==10.0.0
pip-tools==7.3.0
Pygments==2.15.1
pyproject_hooks==1.0.0
requests==2.31.0
tomli==2.0.1
urllib3==2.0.4
watchdog==3.0.0
```

Reply:
Thanks, ok, now let's try comparing homebrew. Here is a list of my Homebrew packages I created by doing `brew bundle dump`. Would you run the same command and send me what you have:
```
tap "dart-lang/dart"
tap "homebrew/bundle"
tap "sass/sass"
brew "autoconf"
brew "automake"
brew "libtool"
brew "pkg-config"
brew "zsh"
brew "sass/sass/sass"
cask "db-browser-for-sqlite"
cask "fontforge"
```

Reply:
Output:
```
tap "homebrew/bundle"
tap "homebrew/cask-fonts"
tap "wix/brew"
brew "adr-tools"
brew "xz"
brew "zstd"
brew "libtiff"
brew "webp"
brew "jpeg-xl"
brew "aom"
brew "openssl@3"
brew "apr-util"
brew "autoconf"
brew "automake"
brew "boost"
brew "libssh2"
brew "curl"
brew "cargo-c"
brew "cmake"
brew "readline"
brew "ruby"
brew "cocoapods"
brew "docker-compose"
brew "folly"
brew "edencommon"
brew "fizz"
brew "wangle"
brew "fbthrift"
brew "fb303"
brew "unbound"
brew "sdl2"
brew "ffmpeg"
brew "libtool"
brew "freetds"
brew "gd"
brew "node"
brew "gitlab-ci-local"
brew "glib-networking"
brew "libpq"
brew "pkg-config"
brew "libsoup"
brew "nvm"
brew "php"
brew "pyenv"
brew "rust"
brew "sdl2_image"
brew "sdl2_mixer"
brew "sdl2_ttf"
brew "tree"
brew "zlib"
brew "wix/brew/applesimutils", args: ["HEAD"]
cask "chromium"
cask "font-fira-code"
cask "font-hack"
cask "font-inconsolata"
```

Reply:
### **Good news!** 
Yes, we can build the Xcode project using the kivy github master. Currently, in the master branch we use `Kivy @ https://github.com/kivy/kivy/archive/master.zip` in the requirements.in file. This works when I update my Homebrew environment with your packages.

Another interesting thing is that pygame is not required after I install your Homebrew packages.

The  _not-so-great news_ is since there are so many differences between our Homebrew environments I’m not sure what package(s) I’m lacking that render me unable to run the app if I don’t install all your Brew packages.

### Path forward
I believe the best path forward is to keep the kivy github master, but add a note in the build_ios.md that kivy==2.2.1 does work and is an option. I say this because if the github master evolves and causes problems in future builds at least we know 2.2.1 is an option. Your thoughts?

### My next question
I will make further updates the build_ios.md file. Should I make another branch (ios_build_02) for those changes or just make the changes to my ios_build_01 and somehow resubmit pull?

The changes briefly:
- include your brew bundle as well and explain how to install 
- I had an error at first attempt to install bundle which was easily remedied using: brew link --overwrite six
- Add note that kivy==2.2.1 does work for the iOS. I think this is good to keep because we don’t know how the kivy github master will evolve.

Reply:
❤️  that's 👌 super news!

- I agree to continue with `kivy master` _**and**_ add an explicit note about `kivy==2.2.1`
- I don't think branching again for other docs changes is required, since this is all the same subject around clarifying the build process for iOS and lists issues and solutions around that build process

Reply:
Ok, the build_ios_01 branch has been updated and I pushed my updates to my forked GitHub repo. Does this require me to renew a pull request?

A quick summary: 
- minor changes made to the build_ios.md. 
- reverted requirements.in back to the exact same state as your master branch, because with the right Homebrew environment setup, it works.

Reply:
👌 just a great job 🙏 !